### PR TITLE
Simpler data file handling

### PR DESCRIPTION
**Pull request by jaraco**
While investigating the purpose behind the `data_files` directive in `setup.py`, I encountered #1538 and #1090. I believe this patch, by dropping custom handling of data files and relying on the tried-and-true techniques for including package data, should fix both of these issues and lessen the debt on the project.

One subtle effect of this change that I want to highlight - the LICENSE.md file will no longer get implicitly copied into the package. I don't believe that's necessary or if it is, it can be handled separately.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -60,35 +59,6 @@
...
-        'cherrypy/test/static/dirback.jpg',
-    ]),
-    ('cherrypy/tutorial', [
-        'cherrypy/tutorial/tutorial.conf',
-        'cherrypy/tutorial/README.txt',
-        'cherrypy/tutorial/pdf_file.pdf',
-        'cherrypy/tutorial/custom_error.html',
-    ]),
-]
 scripts = ['cherrypy/cherryd']
```
Comment (webknjaz) [inline setup.py:62]: Nit: it would be nice to convert this into `console_scripts` entry in `entry_points`

Comment (webknjaz) [reply setup.py:62]: (in separate PR)

File: setup.py [hunk 2]
```diff
@@ -60,35 +59,6 @@
     'cherrypy.scaffold',
     'cherrypy.wsgiserver',
 ]
-data_files = [
-    ('cherrypy', [
-        'cherrypy/cherryd',
-        'cherrypy/favicon.ico',
```
Comment (webknjaz) [inline setup.py:66]: AFAIR `quickstart()` serves favicon by default. Did you test whether this is affected?

File: tests/dist-check.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import os
```
Comment (webknjaz) [inline tests/dist-check.py:15]: This module's name doesn't follow conventions on naming tests.

Comment (jaraco) [reply tests/dist-check.py:15]: I chose the name intentionally not to start with `test_` in order for it not to be automatically collected by pytest and auto-collected. I'll update the module with a docstring explaining why.

Comment (jaraco) [reply tests/dist-check.py:15]: Also, it can't go in cherrypy/test as that causes cherrypy to be imported too early.

File: setup.py (hunk 2 above)
Comment (jaraco) [reply setup.py:66]: Confirmed it's not affected: 

```
$ python -m rwt git+https://github.com/cherrypy/cherrypy@issue-1538 -- -m cherrypy
Collecting git+https://github.com/cherrypy/cherrypy@issue-1538
  Cloning https://github.com/cherrypy/cherrypy (to issue-1538) to /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-ofagwfdc-build
Collecting six (from CherryPy==8.5.1.dev5+ng63f5e94e)
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: six, CherryPy
  Running setup.py install for CherryPy ... done
Successfully installed CherryPy-8.5.1.dev5+ng63f5e94e six-1.10.0
[27/Dec/2016:10:44:07] ENGINE Listening for SIGTERM.
[27/Dec/2016:10:44:07] ENGINE Listening for SIGHUP.
[27/Dec/2016:10:44:07] ENGINE Listening for SIGUSR1.
[27/Dec/2016:10:44:07] ENGINE Bus STARTING
[27/Dec/2016:10:44:07] ENGINE Started monitor thread '_TimeoutMonitor'.
[27/Dec/2016:10:44:07] ENGINE Started monitor thread 'Autoreloader'.
[27/Dec/2016:10:44:07] ENGINE Serving on http://127.0.0.1:8080
[27/Dec/2016:10:44:07] ENGINE Bus STARTED
^Z
zsh: suspended  python3 -m rwt git+https://github.com/cherrypy/cherrypy@issue-1538 -- -m 
$ bg
[1]  + continued  python3 -m rwt git+https://github.com/cherrypy/cherrypy@issue-1538 -- -m 
$ curl http://localhost:8080/favicon.ico
$
```

Comment (webknjaz) [reply setup.py:66]: I've reproduced your steps and found out that it throws `404 Not Found`:
```
$ curl -I http://localhost:8080/favicon.ico
HTTP/1.1 404 Not Found
Connection: close
Date: Tue, 27 Dec 2016 15:48:26 GMT
Server: 127.0.0.1
```
It looks like module you are running isn't using `quickstart` (I didn't really check this).

It's better to check the tutorial's helloworld:
```
$ cd ~/src/cherrypy/cherrypy/tutorial/
$ python tut01_helloworld.py
[27/Dec/2016:17:52:13] ENGINE Listening for SIGUSR1.
[27/Dec/2016:17:52:13] ENGINE Listening for SIGTERM.
[27/Dec/2016:17:52:13] ENGINE Listening for SIGHUP.
[27/Dec/2016:17:52:13] ENGINE Bus STARTING
[27/Dec/2016:17:52:13] ENGINE Started monitor thread 'Autoreloader'.
[27/Dec/2016:17:52:13] ENGINE Started monitor thread '_TimeoutMonitor'.
[27/Dec/2016:17:52:13] ENGINE Serving on http://127.0.0.1:8080
[27/Dec/2016:17:52:13] ENGINE Bus STARTED
^Z
[1]  + 6003 suspended  python tut01_helloworld.py
$ bg
[1]  + 6003 continued  python tut01_helloworld.py
$ curl -I http://localhost:8080/favicon.ico
127.0.0.1 - - [27/Dec/2016:17:52:22] "HEAD /favicon.ico HTTP/1.1" 200 1406 "" "curl/7.46.0"
HTTP/1.1 200 OK
Content-Type: image/x-icon
Content-Length: 1406
Accept-Ranges: bytes
Date: Tue, 27 Dec 2016 15:52:22 GMT
Server: CherryPy/8.3.0.post20161225
Last-Modified: Mon, 25 Jul 2016 19:26:48 GMT
```

File: tests/dist-check.py (hunk 3 above)
Comment (webknjaz) [reply tests/dist-check.py:15]: Alright, then you need to upgrade `.pre-commit-config.yaml` and exclude this file from test name checker :)

File: setup.py (hunk 2 above)
Comment (jaraco) [reply setup.py:66]: Once again I've confirmed that favicon.ico is working, even for the tutorial:

```
$ python -m rwt 'git+https://github.com/cherrypy/cherrypy@issue-1538' -- ~/p/cherrypy/cherrypy/tutorial/tut01_helloworld.py &
[1] 95304
$ Collecting git+https://github.com/cherrypy/cherrypy@issue-1538
  Cloning https://github.com/cherrypy/cherrypy (to issue-1538) to /private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-gdxxok4i-build
Collecting six (from CherryPy==8.5.1.dev6+ng9f8efb28)
  Using cached six-1.10.0-py2.py3-none-any.whl
Installing collected packages: six, CherryPy
  Running setup.py install for CherryPy ... done
Successfully installed CherryPy-8.5.1.dev6+ng9f8efb28 six-1.10.0
[27/Dec/2016:12:16:42] ENGINE Listening for SIGTERM.
[27/Dec/2016:12:16:42] ENGINE Listening for SIGHUP.
[27/Dec/2016:12:16:42] ENGINE Listening for SIGUSR1.
[27/Dec/2016:12:16:42] ENGINE Bus STARTING
[27/Dec/2016:12:16:42] ENGINE Started monitor thread '_TimeoutMonitor'.
[27/Dec/2016:12:16:42] ENGINE Started monitor thread 'Autoreloader'.
[27/Dec/2016:12:16:42] ENGINE Serving on http://127.0.0.1:8080
[27/Dec/2016:12:16:42] ENGINE Bus STARTED

$ curl -I http://localhost:8080/favicon.ico
127.0.0.1 - - [27/Dec/2016:12:16:54] "HEAD /favicon.ico HTTP/1.1" 200 1406 "" "curl/7.51.0"
HTTP/1.1 200 OK
Content-Type: image/x-icon
Server: CherryPy/8.5.1.dev6+ng9f8efb28
Date: Tue, 27 Dec 2016 17:16:54 GMT
Last-Modified: Tue, 27 Dec 2016 17:16:38 GMT
Accept-Ranges: bytes
Content-Length: 1406
```

If you're still getting a 404 (I haven't changed anything), can you dig deeper and identify where the file goes missing in your environment?

Comment (webknjaz) [reply setup.py:66]: I didn't check helloworld using your branch, I just pointed that your previous check was irrelevant :)

File: tests/dist-check.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def test_data_files_installed(data_file_path):
+    """
+    Ensure that data file paths are available in the
+    installed package as expected.
+    """
+    import cherrypy
+    root = os.path.dirname(cherrypy.__file__)
+    fn = os.path.join(root, data_file_path)
```
[the remark this answers is not in the document]
Comment (jaraco) [reply tests/dist-check.py:47]: `fn` is a perfectly fine variable name for a test like this. I can't imagine how to disable this check in a way that's not even more messy or removing the check entirely. I'm not going to worry about it now. It's not worth my time.

File: tests/dist-check.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+"""
```
[the remark this answers is not in the document]
Comment (jaraco) [reply tests/dist-check.py:1]: I specifically named this file not as a typical module name as a signal that it's not meant to be imported. As you can see from the tests which run successfully, it's not invalid for its purpose, which codacy isn't able to ascertain. In any case, I'm not up for fixing it. I'll leave it to those with more free time to change it.

**Pull request merged**